cypress/e2e/permit2.test.ts

```diff
@@ -106,6 +106,28 @@ describe('Permit2', () => {
     cy.get(getTestSelector('popups')).contains('Swapped')
   })
 
+  it('retains modal state when the window loses focus', () => {
+    initiateSwap()
+
+    cy.window().trigger('blur')
+
+    // Verify token approval
+    cy.contains('Enable spending DAI on Uniswap')
```

Comment:
Your wallet will auto-approve this, so I'm afraid that checking for it may be flaky, because it may disappear before cypress sees it. Can you put this under the "(with intermediate screens)" describe block and stubbing and/or turning off auto-mining to guarantee that this is visible and is not flaky.